File: lib/core_services/mixin/auth_mixin.dart
import '../interfaces/user.dart';
import '../service.dart';

mixin AuthMixin on Service {
  static bool support(Object $mixin) {
    return $mixin is AuthMixin && $mixin.$noAuth == false;
  }

  bool get $noAuth => false;
  final bool? $isAuth = true;

  Future<User> getUser({required String cookie});
}

File: lib/core_services/mixin/auth_mixin.ts
import type { User } from "../interfaces/user"
import { BaseService } from "../service"

export abstract class AuthMixin {
  static support($mixin: any): boolean {
    return $mixin instanceof AuthMixin && $mixin.$noAuth === false
  }

  readonly $noAuth: boolean = false
  readonly $isAuth: boolean | null = true
  abstract getUser(params: { cookie: string }): Promise<User>
}
